java/src/main/java/org/rocksdb/WriteOptions.java

```diff
@@ -200,6 +201,86 @@ public boolean lowPri() {
     return lowPri(nativeHandle_);
   }
 
+  /**
+   * If true, this writebatch will maintain the last insert positions of each
+   * memtable as hints in concurrent write. It can improve write performance
+   * in concurrent writes if keys in one writebatch are sequential. In
+   * non-concurrent writes (when {@code concurrent_memtable_writes} is false) this
+   * option will be ignored.
+   *
+   * Default: false
+   *
+   * @return true if writebatch will maintain the last insert positions of each memtable as hints in
+   *     concurrent write.
+   */
+  public boolean memtableInsertHintPerBatch() {
+    return memtableInsertHintPerBatch(nativeHandle_);
+  }
+
+  /**
+   * If true, this writebatch will maintain the last insert positions of each
+   * memtable as hints in concurrent write. It can improve write performance
+   * in concurrent writes if keys in one writebatch are sequential. In
+   * non-concurrent writes (when {@code concurrent_memtable_writes} is false) this
+   * option will be ignored.
+   *
+   * Default: false
+   *
+   * @param memtableInsertHintPerBatch true if writebatch should maintain the last insert positions
+   *     of each memtable as hints in concurrent write.
+   * @return the instance of the current WriteOptions.
+   */
+  public WriteOptions setMemtableInsertHintPerBatch(final boolean memtableInsertHintPerBatch) {
+    setMemtableInsertHintPerBatch(nativeHandle_, memtableInsertHintPerBatch);
+    return this;
+  }
+
+  /**
+   * Timestamp of write operation, e.g. Put. All timestamps of the same
+   * database must share the same length and format. The user is also
+   * responsible for providing a customized compare function via Comparator to
+   * order {@code <key, timestamp>} tuples. If the user wants to enable timestamp, then
+   * all write operations must be associated with timestamp because RocksDB, as
+   * a single-node storage engine currently has no knowledge of global time,
+   * thus has to rely on the application.
+   * The user-specified timestamp feature is still under active development,
+   * and the API is subject to change.
+   *
+   * @return instance to timestamp slice or null if there is no timestamp slice defined.
+   */
+  public Slice timestamp() {
```

Comment:
We plan to merge #8946 early next year which removes `timestamp` from `WriteOptions`. Since this is currently not in Java API, maybe we should avoid adding it so that users do not have to experience addition and deletion of an API in a short period of 2~3 months.

Reply:
@rhubner Could you remove the options for timestamp please? Sorry about that, I was not aware of the planned upcoming change by @riversand963

Reply:
@riversand963 Code updated.